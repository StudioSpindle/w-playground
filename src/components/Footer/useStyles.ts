import { makeStyles, createStyles } from '@material-ui/styles';
import { Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      marginTop: 'auto'
    },
    toolbar: {
      padding: `${theme.spacing(1)} ${theme.spacing(2)}`
    },
    footerItem: {
      marginRight: theme.spacing(2)
    },
    uriRemoveButton: {
      color: theme.palette.common.white
    }
  })
);

export default useStyles;
