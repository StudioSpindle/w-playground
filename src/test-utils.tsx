import React from 'react';
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { MemoryRouter } from 'react-router-dom';
import themeSpec from './themes/default';
import { PlaygroundProvider } from './playground-context';
import '@testing-library/jest-dom/extend-expect';

const AllTheProviders: React.FunctionComponent<{}> = ({ children }) => (
  <PlaygroundProvider>
    <MuiThemeProvider theme={themeSpec.theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </MuiThemeProvider>
  </PlaygroundProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions<Queries>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
