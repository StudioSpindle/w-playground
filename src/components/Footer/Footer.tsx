import React from 'react';
import { Toolbar, Link, Typography, Divider, Box } from '@material-ui/core';
import useStyles from './useStyles';

const Footer: React.FC = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <Divider />
      <Toolbar>
        <Box display="flex" flexDirection="row" flexGrow="1">
          <Box p={1}>
            <Typography>Weaviate Playground BETA</Typography>
          </Box>
          <Box p={1}>
            <Typography>
              <Link href="https://stackoverflow.com/questions/tagged/weaviate" target="_blank">
                request help
              </Link>
            </Typography>
          </Box>
          <Box p={1}>
            <Typography>
              <Link
                href="https://github.com/semi-technologies/weaviate-playground/issues"
                target="_blank"
              >
                report an issue
              </Link>
            </Typography>
          </Box>
          <Box p={1} className={classes.copyright}>
            <Typography>Copyright Â© 2019 SeMI Technologies B.V. All Rights Reserved.</Typography>
          </Box>
        </Box>
      </Toolbar>
    </footer>
  );
};

export default Footer;
