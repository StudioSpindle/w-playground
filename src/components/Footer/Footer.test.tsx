import React from 'react';
import { render } from '../../test-utils';
import Footer from './Footer';

it('matches snapshot', () => {
  const container = render(<Footer />);
  expect(container).toMatchSnapshot();
});
