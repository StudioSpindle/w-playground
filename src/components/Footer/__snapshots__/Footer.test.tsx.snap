// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="makeStyles-footer-1"
      >
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto"
            >
              <p
                class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
              >
                Copyright © 2019 SeMI Technologies B.V. All Rights Reserved.
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
                  >
                    <a
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                      href="https://stackoverflow.com/questions/tagged/weaviate"
                      target="_blank"
                    >
                      request help
                    </a>
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
                  >
                    <a
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                      href="https://github.com/semi-technologies/weaviate-playground/issues"
                      target="_blank"
                    >
                      report an issue
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="makeStyles-footer-1"
    >
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto"
          >
            <p
              class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
            >
              Copyright © 2019 SeMI Technologies B.V. All Rights Reserved.
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
                >
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                    href="https://stackoverflow.com/questions/tagged/weaviate"
                    target="_blank"
                  >
                    request help
                  </a>
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-footerItem-3 MuiTypography-body2"
                >
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                    href="https://github.com/semi-technologies/weaviate-playground/issues"
                    target="_blank"
                  >
                    report an issue
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
